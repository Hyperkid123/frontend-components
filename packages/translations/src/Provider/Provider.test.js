import { render } from '@testing-library/react'

import IntlProvider from './Provider';
describe('provider', () => {
  test('should render correctly', () => {
    const { container } = render(
      <IntlProvider>
        <div>Test</div>
      </IntlProvider>
    );
    expect(container).toMatchSnapshot();
  });

  describe('messages', () => {
    test('no langauge', () => {
      const { container } = render(
        <IntlProvider messages={{ 'some.msg': 'Message' }}>
          <div>Test</div>
        </IntlProvider>
      );
      expect(container).toMatchSnapshot();
    });

    describe('multiple Languages', () => {
      const translations = {
        en: {
          'some.msg': 'Message',
        },
        cs: {
          'some.msg': 'ZprÃ¡va',
        },
      };

      test('no locale', () => {
        const { container } = render(
          <IntlProvider messages={translations}>
            <div>Test</div>
          </IntlProvider>
        );
        expect(container).toMatchSnapshot();
      });

      test('cs-locale', () => {
        const { container } = render(
          <IntlProvider messages={translations} locale="cs">
            <div>Test</div>
          </IntlProvider>
        );
        expect(container).toMatchSnapshot();
      });
    });
  });
});
