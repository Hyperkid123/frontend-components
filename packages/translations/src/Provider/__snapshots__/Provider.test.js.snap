// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provider messages multiple Languages cs-locale 1`] = `
<div>
  <div>
    Test
  </div>
</div>
`;

exports[`provider messages multiple Languages no locale 1`] = `
<div>
  <div>
    Test
  </div>
</div>
`;

exports[`provider messages no langauge 1`] = `
<div>
  <div>
    Test
  </div>
</div>
`;

exports[`provider should render correctly 1`] = `
<div>
  <div>
    Test
  </div>
</div>
`;
